src/Features/Core/Portable/Completion/CompletionOptions.cs

```diff
@@ -12,52 +12,52 @@ namespace Microsoft.CodeAnalysis.Completion
     internal static class CompletionOptions
     {
         // This is serialized by the Visual Studio-specific LanguageSettingsPersister
-        public static readonly PerLanguageOption<bool> HideAdvancedMembers = new PerLanguageOption<bool>(nameof(CompletionOptions), nameof(HideAdvancedMembers), defaultValue: false);
+        public static readonly PerLanguageOption2<bool> HideAdvancedMembers = new PerLanguageOption2<bool>(nameof(CompletionOptions), nameof(HideAdvancedMembers), defaultValue: false);
```

Comment:
Are any of these option types being used by TypeScript or F# IVT?

Reply:
I ensured the ones used by F# in their external access layer have the public API type. I need to check that the TS ones are not broken.

Reply:
Thanks for this reminder. 

**Used via IVT (unchanged in this PR)**
CompletionOptions: Used by Editor + TypeScript
PythiaOptions: Used by Pythia - I had changed these options but now reverted it
FSharpNavigationOptions, FSharpServiceFeatureOnOffOptions, FSharpCompletionOptions: All in external access layer.

**Used via IVT (changed in this PR - verified that option directly passed to WithChangedOption/GetOption, both of which now have internal overloads for Option2 and PerLanguageOption2)**
EditorComponentOnOffOptions: Used by Editor
NavigationOptions: Used by RoslynTaoActions
ServiceFeatureOnOffOptions: Used by TypeScript

**Public types (unchanged in this PR)**
RenameOptions
FormattingOptions

Reply:
> Used via IVT (changed in this PR - verified that option directly passed to WithChangedOption/GetOption, both of which now have internal overloads for Option2 and PerLanguageOption2)

Is this still going to break though? It's not a source breaking change but would be a binary breaking change since the TypeScript code would directly be binding to the return that returned an PerLanguageOption<>?

Reply:
Sounds reasonable - I will revert these changes.